taskcluster/client.py

```diff
@@ -67,6 +74,17 @@ def __init__(self, options=None, session=None):
                     except:
                         s = '%s (%s) must be unicode encodable' % (x, credentials[x])
                         raise exceptions.TaskclusterAuthFailure(s)
+
+        baseUrl = o.get('baseUrl')
+        rootUrl = o.get('rootUrl')
+
+        if rootUrl and not o.get('version'):
+            s = 'specifying a rootUrl requires a name and version in the API reference'
+            raise exceptions.TaskclusterFailure(s)
+        elif rootUrl:
+            name = o.get('name', urllib.parse.urlparse(baseUrl).netloc.split('.', 1)[0])
+            o['baseUrl'] = '%s/%s/v%s' % (rootUrl, name, o['version'])
```

Comment:
This probably ought to be something like `%s/api/%s/v%s` which we define in taskcluster-lib-urls. I actually just got that python implementation up for review in taskcluster/taskcluster-lib-urls#6 if you want to take a look. Once that is r+, we can use it here to build urls.

Reply:
This has now landed! https://github.com/taskcluster/taskcluster-lib-urls#python-usage